pkg/services/extsvcauth/extsvcaccounts/service.go

```diff
@@ -118,9 +166,75 @@ func (esa *ExtSvcAccountsService) saveExtSvcAccount(ctx context.Context, cmd *sa
 
 // deleteExtSvcAccount deletes a service account by ID and removes its associated role
 func (esa *ExtSvcAccountsService) deleteExtSvcAccount(ctx context.Context, orgID int64, slug string, saID int64) error {
-	esa.logger.Info("Delete service account", "service", slug, "saID", saID)
+	esa.logger.Info("Delete service account", "service", slug, "orgID", orgID, "saID", saID)
 	if err := esa.saSvc.DeleteServiceAccount(ctx, orgID, saID); err != nil {
 		return err
 	}
-	return esa.acSvc.DeleteExternalServiceRole(ctx, slug)
+	if err := esa.acSvc.DeleteExternalServiceRole(ctx, slug); err != nil {
+		return err
+	}
+	return esa.DeleteExtSvcCredentials(ctx, orgID, slug)
+}
+
+// getExtSvcAccountToken get or create the token of an External Service
+func (esa *ExtSvcAccountsService) getExtSvcAccountToken(ctx context.Context, orgID, saID int64, extSvcSlug string) (string, error) {
+	// Get credentials from store
+	credentials, err := esa.GetExtSvcCredentials(ctx, orgID, extSvcSlug)
+	if err != nil && !errors.Is(err, extsvcauth.ErrCredentialsNotFound) {
+		return "", err
+	}
+	if credentials != nil {
+		return credentials.Secret, nil
+	}
+
+	// Generate token
+	esa.logger.Info("Generate new service account token", "service", extSvcSlug, "orgID", orgID)
```

Comment:
if this is logged every time a token is retrieved maybe should be `Debug`

Reply:
It's logged only the first time the token is created :+1:. Afterwards, the token being retrieved from the secret store, we won't re-generate it.